File: xzcpc-xzg/people/src/main/java/com/xzcpc/people/entity/OwnerRegistration.java
package com.xzcpc.people.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("owner_registration")
public class OwnerRegistration {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String openid;
    private String bindCode;
    private String name;
    private String phone;
    private String storeId;
    private String storeName;
    private String role;
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: admin/src/api/ownerRegistration.ts
import api from './index'

export interface OwnerRegistration {
  id: number
  openid: string
  bindCode?: string
  name: string
  phone: string
  storeId: string
  storeName: string
  role: string
  status: string
  createdAt: string
}

export function getOwnerRegistrations(params?: {
  storeId?: string
  status?: string
  name?: string
  phone?: string
  pageNum?: number
  pageSize?: number
}) {
  return api.get('/owner-registrations', { params })
}

export function bindOwnerRegistration(id: number, data: { name: string; phone: string; role: string }) {
  return api.post(`/owner-registrations/${id}/bind`, data)
}

export function updateOwnerRegistration(id: number, data: { name: string; phone: string; role: string }) {
  return api.put(`/owner-registrations/${id}`, data)
}
